dll.py

```diff
@@ -0,0 +1,111 @@
+#! /usr/bin/env python
+# Queue Python file for Efrain, Mark and Henry
+
+
+class Node(object):
+    """A Node Class representing a Node in a queue.
+    Each node has a pointer to the next node and a value.
+    """
+
+    def __init__(self, value, back=None, front=None ):
+        """Create Node with value and optional pointer.
+        If no pointer is specified, pointer is set to None.
+        """
+        self.previous = back
+        self.value = value
+        self.next = front
+
+
+class Dll(object):
+    """Class defining a dll data structure."""
+
+    def __init__(self):
+        """Create a dll with a pointer to the front and back of an empty dll
+        and set count to zero to keep track of size"""
+        self.head = None
+        self.tail = None
+        self.count = 0
+
+    def append(self, value):
+        """Adds a node with the value to the tail of the dll"""
+        old_tail = self.tail
+        self.tail = Node(value, None, old_tail)
+        if self.count > 0:  # if any Nodes: set tail previous to current Node
+            old_tail.previous = self.tail
+        else:  # adding to an empty, than define front
+            self.head = self.tail
+        self.count += 1
+
+    def insert(self, value):
+        """Adds a node with the value to the front of the dll"""
```

Comment:
I would suggest saying it adds to the head rather than front - just so there's no confusion on what this method does